opentelemetry-instrumentation/src/opentelemetry/instrumentation/dependencies.py

```diff
@@ -25,20 +25,27 @@ def __str__(self):
 def get_dist_dependency_conflicts(
     dist: Distribution,
 ) -> Optional[DependencyConflict]:
-    deps = [
-        dep
-        for dep in dist.requires(("instruments",))
-        if dep not in dist.requires()
-    ]
-    return get_dependency_conflicts(deps)
+    main_deps = dist.requires()
+    instrumentation_deps = []
+    for dep in dist.requires(("instruments",)):
```

Comment:
Just to clarify, this would only generate the `Requirements` for the underlying instrumented library correct?

Reply:
Correct. pkg_resource does not have an API to fetch only a specific extra require group. We could have used something like `dist.requires(only_extra=('instruments',))` but it doesn't exist :)